### ISSUE TITLE
[BUG]-FieldMapping uses target data instead of source data

### ISSUE THREAD

**Issue by AlwaysThinkin** [forcedotcom/SFDX-Data-Move-Utility#663]
**Describe the bug**
When performing a data migration upsert operation with field mappings and skipExistingRecords set to false, the fieldmapping uses the data from the target to update the target if you are mapping the source org's record Id to a field in the target org.

**To Reproduce**

1. Create a text field on Lead in the target org called Source Record Id
2. Run a migration with an upsert operation, using field mappings of Lead,Id,Source_Record_Id__c
3. Confirm that the source org Lead Id has populated the Source Record Id (as expected)
4. Run the same migration with an upsert operation, using field mappings, and setting skipExistingRecords: false so that updates are possible.
5. Review the Source_Record_Id__c field and note that the original source org Lead Id has been replaced with the target org's Lead Id.

**Expected behavior**
I expect that when an update is performed with field mappings for the source org Id field to a target org field that the Id from the source org is written to the target org. Other source org field data is written to the target org as expected but the Id field appears to be an exception.

**export.json**
[export.json](https://github.com/forcedotcom/SFDX-Data-Move-Utility/files/13370977/export.json)
[FieldMapping.csv](https://github.com/forcedotcom/SFDX-Data-Move-Utility/files/13370978/FieldMapping.csv)
[2023-11-15__19_21_55.log](https://github.com/forcedotcom/SFDX-Data-Move-Utility/files/13370979/2023-11-15__19_21_55.log)
[2023-11-15__19_23_16.log](https://github.com/forcedotcom/SFDX-Data-Move-Utility/files/13370980/2023-11-15__19_23_16.log)
[Lead_update_target.csv](https://github.com/forcedotcom/SFDX-Data-Move-Utility/files/13370981/Lead_update_target.csv)

**Comment by hknokh2** [comment]
Hello,
Thank you for reaching out to me. 
I will take a look at your case and let you know if there are any updates.
Cheers.

**Comment by hknokh2** [comment]
Hello,

I've looked into your case. The issue you mentioned, where the target mapped field is replaced with the target ID when `skipExistingRecords = false`, was not replicated in my tests. 

However, I did identify and fix a different issue. 
When `skipExistingRecords = false`, the tool wasn't updating the `Source_Record_Id__c` field on the target record in cases where all fields of the source record were identical except for `Source_Record_Id__c` (which might have been empty, for instance). This field should have been populated but wasn't, and this issue has now been resolved.

It's important to note the purpose of setting `skipExistingRecords = true`. This setting is intended to prevent updates to already existing target records (which are compared by their external ID value with the respective source records). 
Therefore, when you set `skipExistingRecords = true` and `operation = Upsert`, the tool will not update existing records. It will only insert new ones, meaning the `Source_Record_Id__c` will never be updated for existing records. However, it will correctly populate `Source_Record_Id__c` with the source ID values for new target records.

Best regards.
